lua/starfall/libs_sv/prop.lua

```diff
@@ -430,9 +458,48 @@ function props_library.createSent(pos, ang, class, frozen)
 		end
 
 		hookcall = "PlayerSpawnedVehicle"
-
+	elseif sent2 then
+		if ply ~= NULL then
+			if (scripted_ents.GetStored(class).t.AdminOnly and not ply:IsAdmin()) then SF.Throw("This sent is admin only!", 2) end
+			if gamemode.Call("PlayerSpawnSENT", ply, class) == false then SF.Throw("Another hook prevented the sent from spawning", 2) end
+		end
+		
+		local enttbl = {}
+		data = data or {}
+		
+		-- Apply data
+		for param, org in pairs(sent2[1]) do
+			local value = data[param]
+			
+			if value then
+				checkluatype(value, org[2])
```

Comment:
Better error message here will be needed. Maybe check type TypeID(value)==org[2] and if not then SF.ThrowTypeError https://github.com/thegrb93/StarfallEx/blob/master/lua/starfall/libs_sh/builtins.lua#L756